### PR TITLE
feat(migration): #MEX-369 migration Vertx

### PR DESCRIPTION
**Pull request by Alex-Delaveau**
## Describe your changes
Migration from vertx3 to vertx4
## Checklist tests
Try to compile and launch the module in your springboard
## Issue ticket number and link
[MEX-369](https://jira.support-ent.fr/browse/MEX-369)
## Checklist before requesting a review (magic string, indentation, comment/documentation...)

- [x] I have detailed the tests to do in my feature/fix in order to prevent consequents regressions (must specify in **Checklist tests**)
- [x] I have performed a self-review of my code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation (API Doc etc...) - (must specify in **Description** for target version)
- [ ] If it is a consequent feature, I have added thorough tests.
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] Any dependent changes have been added to this project (must specify in **Description**)

### CODE REVIEW

File: src/main/java/fr/openent/pmb/helper/HttpClientHelper.java [hunk 1]
```diff
@@ -70,42 +68,50 @@ public static void webServicePmbGet(String pmbUrl, HttpServerRequest request,
...
+
+
+        httpClient.request(requestOptions)
+                .flatMap(clientRequest ->  {
+                    clientRequest.setFollowRedirects(true);
+                    return clientRequest.send();
+                })
+                .onSuccess(response -> {
+                    if (response.statusCode() != 200) {
+                        log.error("Fail to get webservice" + response.statusMessage());
```
Comment (qperie) [inline src/main/java/fr/openent/pmb/helper/HttpClientHelper.java]:
```suggestion
                        log.error("[PMB@HttpClientHelper::webServicePmbGet] Fail to get webservice" + response.statusMessage());
```

File: src/main/java/fr/openent/pmb/helper/HttpClientHelper.java [hunk 2]
```diff
@@ -70,42 +68,50 @@ public static void webServicePmbGet(String pmbUrl, HttpServerRequest request,
...
+                        response.bodyHandler(event -> {
+                            log.error("Returning body after GET CALL : " + pmbUrl + ", Returning body : " + event.toString("UTF-8"));
```
Comment (qperie) [inline src/main/java/fr/openent/pmb/helper/HttpClientHelper.java]:
```suggestion
                            log.error("[PMB@HttpClientHelper::webServicePmbGet] Returning body after GET CALL : " + pmbUrl + ", Returning body : " + event.toString("UTF-8"));
```

File: src/main/java/fr/openent/pmb/helper/HttpClientHelper.java [hunk 3]
```diff
@@ -70,42 +68,50 @@ public static void webServicePmbGet(String pmbUrl, HttpServerRequest request,
...
+                    response.handler(body::appendBuffer);
+                    response.endHandler(end -> {
+                        handler.handle(new Either.Right<>(body));
+                        if (!responseIsSent.getAndSet(true)) {
+                            httpClient.close();
+                        }
+                    });
+                })
+                .onFailure(throwable -> {
+                    log.error(throwable.getMessage(), throwable);
```
Comment (qperie) [inline src/main/java/fr/openent/pmb/helper/HttpClientHelper.java]:
```suggestion
                    log.error("[PMB@HttpClientHelper::webServicePmbGet] " + throwable.getMessage(), throwable);
```

File: src/main/java/fr/openent/pmb/server/PMBServer.java [hunk 4]
```diff
@@ -90,20 +87,25 @@ private String uri(String dbName) {
...
+                    if (response.statusCode() != 200) {
+                        handler.handle(Future.failedFuture(response.statusMessage()));
+                        return;
+                    }
+                    Buffer body = Buffer.buffer();
+                    response.bodyHandler(body::appendBuffer);
+                    response.endHandler(aVoid -> handler.handle(Future.succeededFuture(new JsonObject(new String(body.getBytes())))));
+                    response.exceptionHandler(throwable -> handler.handle(Future.failedFuture(throwable)));
+                })
+                .onFailure(throwable -> handler.handle(Future.failedFuture(throwable)));
```
Comment (qperie) [inline src/main/java/fr/openent/pmb/server/PMBServer.java]: Ajouter un log d'erreur

File: src/main/java/fr/openent/pmb/server/PMBServer.java [hunk 5]
```diff
@@ -90,20 +87,25 @@ private String uri(String dbName) {
...
+        RequestOptions requestOptions = new RequestOptions()
+                .setAbsoluteURI(this.uri(uai))
+                .setMethod(HttpMethod.POST)
+                .addHeader(HttpHeaders.CONTENT_TYPE.toString(), "application/json")
+                .addHeader(HttpHeaders.AUTHORIZATION.toString(), String.format("Basic %s", credential.basic()));
+
+        httpClient.request(requestOptions)
+                .flatMap(request -> request.send(content.encode()))
+                .onSuccess(response -> {
+                    if (response.statusCode() != 200) {
```
Comment (qperie) [inline src/main/java/fr/openent/pmb/server/PMBServer.java:99]: Ajouter un log d'erreur

**Review by qperie — CHANGES_REQUESTED** [review]

**Review by Glama107 — APPROVED** [review]

**Review by JinDouang — APPROVED** [review]

**Pull request merged**